tensorflow/compiler/xla/service/elemental_ir_emitter.cc

```diff
@@ -454,75 +454,47 @@ StatusOr<llvm::Value*> ElementalIrEmitter::EmitCusUnaryOp(
           primitive_util::BitWidth(from_type),
           primitive_util::BitWidth(to_type));
     }
-    // case HloOpcode::kExp:
-    //   return EmitExp(op->shape().element_type(), operand_value);
+    case HloOpcode::kExp:{
+      // todo(chenhao) cast to f32 to calculate for now
+      operand_value = EmitCusToF32(operand_value, b_);
+      auto value = EmitExp(op->shape().element_type(), operand_value);
+      return EmitF32ToCus(value.ValueOrDie(), b_);
```

Comment:
Instead of emitting the casting operation here, can you do the casting inside the exp function defined under struct cus?

Reply:
Done